pages/content/amp-dev/documentation/guides-and-tutorials/optimize-measure/amp_to_pwa.md

```diff
@@ -0,0 +1,204 @@
+AMP_SW.init();
+[/sourcecode]
+
+With just two lines of code, this imports the AMP Service Worker into your Service Worker and initializes it.
+
+## Auto-install your service worker on your AMP pages
+
+AMP websites use the [`<amp-install-serviceworker>`](https://www.ampproject.org/docs/reference/components/amp-install-serviceworker) 
+component to install the service worker in the browser's background, while the user is enjoying your content.
+
+Place the required script tag in the head of `index.html` and the `<amp-install-serviceworker>` element inside the `<body>`:
+
+[sourcecode:html]
+…
+<script async custom-element="amp-install-serviceworker" src="https://cdn.ampproject.org/v0/amp-install-serviceworker-0.1.js"></script>
+…
+</head>
+<body>
+   <amp-install-serviceworker src="/sw.js"
+           data-iframe-src="install-sw.html"
+           layout="nodisplay">
+   </amp-install-serviceworker>
+...
+[/sourcecode]
+
+
+The `<amp-install-serviceworker>` installs the service worker by creating an iframe and running the `data-iframe-src` file. 
+Create the `install-sw.html` file and add the following code:
+
+[sourcecode:html]
+<!doctype html>
+<title>installing service worker</title>
+<script type='text/javascript'>
+ if('serviceWorker' in navigator) {
+   navigator.serviceWorker.register('./sw.js');
+ };
+</script>
+[/sourcecode]
+
+The iframe registers the AMP Service Worker file into the browser. 
+
+
+# Customize what is cached 
+
+The AMP Service Worker comes with built-in benefits while allowing optional fields that you can configure to optimize against your app's needs. 
+
+Our music festival app will cache our image assets, prefetch the line up link, and specify an offline page. 
+
+
+## Cache Assets
+
+You can configure the AMP Service Worker to [cache assets](https://github.com/ampproject/amp-sw/tree/master/src/modules/asset-caching), 
+such as images, videos and fonts. We'll use it to cache our background image and the AMP logo. Open the `sw.js` file and update it to the code below:
+
+[sourcecode:js]
+importScripts('https://cdn.ampproject.org/sw/amp-sw.js');
+AMP_SW.init({
+   assetCachingOptions: [{
+       regexp: /\.(png|jpg)/,
+       cachingStrategy: 'CACHE_FIRST'
+     }]
+});
+[/sourcecode]
+
+
+We've specified the caching strategy to be [cache first](https://developers.google.com/web/fundamentals/instant-and-offline/offline-cookbook/#cache-falling-back-to-network). 
+The means the app will try to serve images from cache first before requesting anything from the network. This is especially useful for this app since we won't be updating our background image or the AMP logo. 
+
+
+## Prefetch Links
+
+The AMP Service Worker prefetches links that have the `data-rel=prefetch` attribute. 
+This enables users to view pages offline even if they haven't visited them yet. We'll add the attribute to our link tag for `lineup.html`.
+
+
+[sourcecode:html]
+...
+<a href="/lineup.html" data-rel="prefetch">See Full Lineup</a>
```

Comment:
sidenote: we should use this approach for step-by-step guides in amp.dev

Reply:
What part of the approach? All on one page?